gpkit/constraints/prog_factories.py

```diff
@@ -0,0 +1,115 @@
+"Scripts for generating, solving and sweeping programs"
+import numpy as np
+from time import time
+from ..nomials.substitution import parse_subs
+from ..solution_array import SolutionArray
+from ..keydict import KeyDict
+from ..varkey import VarKey
+from ..nomials import Monomial
+
+try:
+    from ipyparallel import Client
+    CLIENT = Client(timeout=0.01)
+    assert len(CLIENT) > 0
+    POOL = CLIENT[:]
+    POOL.use_dill()
+    print("Using parallel execution of sweeps on %s clients" % len(CLIENT))
+except (ImportError, IOError, AssertionError):
+    POOL = None
+
+
+def _progify_fctry(program, return_attr=None):
+    "Generates function that returns a program() and optionally an attribute."
+    def programify(self, verbosity=1, substitutions=None):
+        if not substitutions:
+            substitutions = self.substitutions
```

Comment:
else... should an update happen? is `self.substitutions` being ignored if `substitutions` is not None?